src/network/mod.rs

```diff
@@ -0,0 +1,55 @@
+// Crate contains the types which are accepted by netvark.
+
+extern crate serde_derive;
+use std::collections::HashMap;
+
+// NetworkOptions for a given container.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct NetworkOptions {
+    #[serde(rename = "container_id")]
+    pub container_id: Option<String>,
+
+    #[serde(rename = "container_name")]
+    pub container_name: Option<String>,
+
+    #[serde(rename = "networks")]
+    pub networks: Option<HashMap<String, PerNetworkOptions>>,
+
+    #[serde(rename = "port_mappings")]
+    pub port_mappings: Option<Vec<PortMapping>>,
+}
+
+// PerNetworkOptions are options which should be set on a per network basis
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PerNetworkOptions {
+    #[serde(rename = "aliases")]
+    pub aliases: Option<Vec<String>>,
+
+    #[serde(rename = "interface_name")]
+    pub interface_name: Option<String>,
```

Comment:
Interface name should not be optional. I assume the `Option` means that this can be empty or what does this do in rust?

Reply:
@Luap99 this wraps the value so `None` can be parsed this does not necessarily means that this is optional. We can enforce mandatory checks in business logic struct shouldn't specify that.

Reply:
Thanks for clarifying this. I do not really understand the rust syntax yet and this wasn't clear to me from the rust documentation.

Reply:
Is there a reason that container_port does not use Optional?

Reply:
@Luap99 Since `containerPort` is missing omitempty here https://github.com/containers/podman/blob/main/libpod/network/types/network.go#L184 it is assumed that this will be always present and can never be `None`

Reply:
@Luap99 We can remove `Option<` from eveything but then it would mean that all the values have to be filled by some values they cannot be `None`.

Reply:
@Luap99 Could you remove `omitempty` from go types which are mandatory ?

Reply:
@flouthoc Yes I will do this.

Reply:
Updated after https://github.com/containers/podman/pull/11692